clients/src/main/java/org/apache/kafka/clients/admin/MemberDescription.java

```diff
@@ -21,24 +21,40 @@
 
 import java.util.Collections;
 import java.util.Objects;
+import java.util.Optional;
 
 /**
  * A detailed description of a single group instance in the cluster.
  */
 public class MemberDescription {
     private final String memberId;
+    private final Optional<String> groupInstanceId;
     private final String clientId;
     private final String host;
     private final MemberAssignment assignment;
 
-    public MemberDescription(String memberId, String clientId, String host, MemberAssignment assignment) {
+    @Deprecated
+    public MemberDescription(String memberId,
```

Comment:
Actually the constructor of all such description classes can just be default package-private since they are only used by KafkaAdminClient, and hence can just be private APIs, and we do not need to deprecate-overload any more.